addons/binding/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/internal/kostal/inverter/ThirdGeneration/ThirdGenerationMappingInverterToChannel.java

```diff
@@ -0,0 +1,1098 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.internal.kostal.inverter.thirdgeneration;
+
+import static org.openhab.binding.internal.kostal.inverter.thirdgeneration.ThirdGenerationBindingConstants.*;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * The {@link ThirdGenerationMappingInverterToChannel} is responsible for the management of the channels
+ * available to inverters
+ *
+ * @author René Stakemeier - Initial contribution
+ */
+class ThirdGenerationMappingInverterToChannel {
+
+    private static Map<ThirdGenerationInverterTypes, List<ThirdGenerationChannelMappingToWebApi>> channelMapping = new HashMap<ThirdGenerationInverterTypes, List<ThirdGenerationChannelMappingToWebApi>>();
+
+    /*
+     * Assign the channels to the devices.
+     */
+    static {
+        /*
+         * Channels available on all devices
+         */
+        addInverterChannel(
```

Comment:
The channel setup below looks very repetitive? Are you sure this can not be expressed better?

Reply:
I could reduce the lines of code by using local variables